tests/test_door_control.py

```diff
@@ -0,0 +1,450 @@
+                {
+                    "token": "Axis-cbb9614a-445d-45b3-9abf-59a6ebe593a3",
+                    "Name": "Main Door",
+                    "Description": "",
+                    "Capabilities": {
+                        "Access": True,
+                        "Lock": True,
+                        "Unlock": True,
+                        "Block": True,
+                        "DoubleLock": True,
+                        "LockDown": True,
+                        "LockOpen": True,
+                        "DoorMonitor": True,
+                        "LockMonitor": False,
+                        "DoubleLockMonitor": False,
+                        "Alarm": True,
+                        "Tamper": False,
+                        "Warning": True,
+                        "Configurable": True
+                    }
+                }
+            ]
+        }
+    )
+
+    tokens = ["Axis-cbb9614a-445d-45b3-9abf-59a6ebe593a3"]
+
+    response = await door_control.get_door_info(tokens)
+
+    assert route.called
+    assert route.calls.last.request.method == "POST"
+    assert route.calls.last.request.url.path == "/vapix/doorcontrol"
+    assert json.loads(route.calls.last.request.content) == {
+        "tdc:GetDoorInfo": {"Token": tokens}
+    }
+
+    assert response["DoorInfo"] == [
+        {
+            "token": "Axis-cbb9614a-445d-45b3-9abf-59a6ebe593a3",
+            "Name": "Main Door",
+            "Description": "",
+            "Capabilities": {
+                "Access": True,
+                "Lock": True,
+                "Unlock": True,
+                "Block": True,
+                "DoubleLock": True,
+                "LockDown": True,
+                "LockOpen": True,
+                "DoorMonitor": True,
+                "LockMonitor": False,
+                "DoubleLockMonitor": False,
+                "Alarm": True,
+                "Tamper": False,
+                "Warning": True,
+                "Configurable": True
+            }
+        }
+    ]
+
+
+@respx.mock
+async def test_get_door_state(door_control):
+    """Test get door state(s)."""
+    route = respx.post(f"http://{HOST}:80/vapix/doorcontrol").respond(
+        json={
+            "DoorState": {
+                "DoorPhysicalState": "Closed",
+                "Alarm": "Normal",
+                "DoorMode": "Locked"
+            }
+        }
+    )
+
+    token = "Axis-cbb9614a-445d-45b3-9abf-59a6ebe593a3"
+
+    response = await door_control.get_door_state(token)
+
+    assert route.called
+    assert route.calls.last.request.method == "POST"
+    assert route.calls.last.request.url.path == "/vapix/doorcontrol"
+    assert json.loads(route.calls.last.request.content) == {
+        "tdc:GetDoorState": {"Token": token}
+    }
+
+    assert response["DoorState"] == {
+        "DoorPhysicalState": "Closed",
+        "Alarm": "Normal",
+        "DoorMode": "Locked"
+    }
+
+
+@respx.mock
+async def test_access_door(door_control):
```

Comment:
All tests after this point are 99% equal, could you use parametrize?

https://github.com/Kane610/axis/blob/13560f755e4043a12c63b40a1be793d39cd2bde6/tests/test_event_stream.py#L53

Reply:
Yep, will do

Reply:
Got it :-)